### ISSUE TITLE
CloudFormation ref in Fn:GetAtt and Fn:FindInMap shows validations errors Incorrect type. Expected "string". yaml-schema: aws://cloudformation.schema.json

### ISSUE THREAD

**Issue by lirem** [redhat-developer/vscode-yaml#867]
## Describe the bug
I have "Incorrect type. Expected "string".yaml-schema: aws://cloudformation.schema.json" error, but aws is working normally with this template.
<img width="281" alt="Screenshot 2023-01-06 at 11 26 41" src="https://user-images.githubusercontent.com/75673262/210991128-cbd80610-046e-4674-b62e-e82e8c91b441.png">
<img width="884" alt="Screenshot 2023-01-06 at 11 26 04" src="https://user-images.githubusercontent.com/75673262/210991145-c6737466-d79f-4f98-97af-7ba82c02cf7b.png">

## Expected Behavior
Nothing shoud be highlighted, because it's working template

## Environment
- VsCode mac

**Comment by citrix-eg** [comment]
Similar problem. And to add to this I have added all of the yaml.customTags for each of other intrinsic functions of CloudFormation.

**Comment by lirem** [comment]
Yeah, the best way when the YAML extension still doesn't work is to use the CloudFormation Linter extension. It works very well with CloudFormation and even helps you with some errors and gives you warnings(e.g. defined but not used resources)

**Comment by allenfisher** [comment]
I'm seeing this as well. I'd really like this fixed, I like to use the CFN extension provided by MS for some things.

**Comment by paulJRCurtis** [comment]
I added the below to my vscode settings and it solved most of these issues.

```
"yaml.customTags": [
	"!GetAtt sequence",
	"!GetAtt scalar",
	"!Ref scalar",
	"!Sub scalar",
	"!GetAZs scalar",
	"!Select sequence"
]
```

I use the CloudFormation short form so the above works for me but you'd need to change to the long form based on your example. That and for !GetAtt if you use the short form syntax it works.

`AllocationId: !GetAtt PublicSubnet1EIP.AllocationId`

I have yet to work out how to deal with `AvailabilityZone: !Select [ 1, !GetAZs  '' ]` returning a `Incorrect type. Expected "string"` error.

**Comment by TeeSeal** [comment]
I am having a similar problem with:
```yaml
  Api:
    Type: "AWS::Serverless::Api"
    Properties:
      Name: !Join ["-", ["foo", !Ref "bar"]]
```
I understand that it is not possible to infer the return type from custom tags at this time. Perhaps a way to define said return type would help? Or otherwise a way to disable validations where custom tags are used would be nice.

**Comment by fushihara** [comment]
I gave up and
I disabled the validator by setting “yaml.validate”: false, in \.vscode\settings.json
It changes the color of the key and value text, so it's still better than editing directly in Notepad!

**Comment by rotty3000** [comment]
I feel like being able to specify a return type for tags would save a tone of errors with intrinsic functions and make them a lot more friendly.

**Comment by shin19991207** [comment]
This issue has been resolved in https://github.com/redhat-developer/yaml-language-server/pull/1255, which introduces support for optional return types in the `yaml.customTags` setting.

`yaml.customTags` now supports three forms:
```
!Tag
!Tag nodeType
!Tag nodeType:returnType
```
In the new format, "!Tag nodeType:returnType", nodeType specifies the YAML node type that the tag is written on, and returnType specifies the schema type that the tagged value evaluates to. Existing configurations remain supported, so this is backward compatible. Behavior only changes when a user explicitly adds :returnType.

Note that `yaml.customTags` only applies to YAML short-form tags. It does not apply to CloudFormation long-form syntax. Long-form `Fn::...` values are normal YAML mappings, not YAML tags. 

### CloudFormation / SAM Usage

For users working with AWS CloudFormation or SAM templates, below is the updated configuration for all intrinsic functions in short form, with return types included, based on the [AWS CloudFormation intrinsic function docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html):

```json
{
  "yaml.customTags": [
    "!Base64 scalar:string",
    "!Cidr sequence:array",
    "!And sequence:boolean",
    "!Equals sequence:boolean",
    "!If sequence",
    "!Not sequence:boolean",
    "!Or sequence:boolean",
    "!Condition scalar:boolean",
    "!FindInMap sequence:string",
    "!GetAtt scalar:string",
    "!GetAtt sequence:string",
    "!GetAZs scalar:array",
    "!ImportValue scalar:string",
    "!Join sequence:string",
    "!Select sequence:string",
    "!Split sequence:array",
    "!Sub scalar:string",
    "!Sub sequence:string",
    "!Transform mapping",
    "!Ref scalar:string"
  ]
}
```
